src/sentry/issues/escalating.py

```diff
@@ -154,36 +167,167 @@ def _query_with_pagination(
 ) -> List[GroupsCountResponse]:
     """Query Snuba for event counts for the given list of project ids and groups ids in
     a time range."""
+    organization = Organization.objects.get(id=organization_id)
     all_results = []
     offset = 0
+
     while True:
-        query = _generate_query(project_ids, group_ids, offset, start_date, end_date, category)
+        query = _generate_entity_dataset_query(
+            project_ids, group_ids, offset, start_date, end_date, category
+        )
         request = Request(
             dataset=_issue_category_dataset(category),
             app_id=REFERRER,
             query=query,
             tenant_ids={"referrer": REFERRER, "organization_id": organization_id},
         )
         results = raw_snql_query(request, referrer=REFERRER)["data"]
+
         all_results += results
         offset += ELEMENTS_PER_SNUBA_PAGE
         if not results or len(results) < ELEMENTS_PER_SNUBA_PAGE:
             break
 
+    if category == GroupCategory.ERROR and features.has(
+        "organizations:escalating-issues-v2", organization
+    ):
+        metrics_results = []
```

Comment:
Can you please put this if-block into its own method? It will reduce the length of `_query_with_pagination` and readability.